eclipsesource/theia
# Ensure that the Debug API is working with the new RPC

#### What it does
After migrating to new binary RPC (https://github.com/eclipse-theia/theia/commit/83d430846182a2ccd8465bcfd28ba7214c42abf3) Debugging is no longer working as expected.
The main reason for this an incomplete migration of the Debug API to the new binary message channels. In addition, to being incomplete this refactoring is also unnecessary. The Debug (Adapter) Protocol is completely string based anyways so there is no benefit in migrating to the binary protocol. So this PR fixes Debugging in Theia by
- Restoring the original  string-based message channel for Debug communication. 
- Rename the string-based `Channel` used in the Debug API to `DebugChannel` to avoid naming confusion with the default binary message `Channel`s.
- Only use rely on binary channels where absolutely necessary and introduce a `ForwardingDebugChannel` which can be used to wrap an arbitrary binary message `Channel` (i.e. the service websocket channel`) in a `DebugChannel`

Fixes #11263

Contributed on behalf of STMicroelectronics


#### How to test

1. Open a .spec.ts file in the repo and try the `Run Mocha Tests` debug configuration.
2. Debug view should work as expected. (Variables are shown, debug, debug trace, breakpoins are hit etc.
3 Try a dynamic Node configuration like 'Run current file'. Should also work as expected.

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
